backend-api/src/functions/services/session/tests/sessionService.test.ts

```diff
@@ -0,0 +1,304 @@
+        dynamoDbMockClient
+          .on(QueryCommand)
+          .resolvesOnce({ Items: [{ sessionId: { S: "mockSessionId" } }] });
+
+        const result = await sessionService.getActiveSession("mockSub");
+
+        expect(result.isError).toBe(true);
+        expect(result.value).toEqual({
+          errorCategory: "SERVER_ERROR",
+          errorMessage: "Session is malformed",
+        });
+      });
+    });
+
+    describe("Given the item returned does not have the attribute redirectUri", () => {
+      it("Returns a success response with the session ID and state as value", async () => {
+        dynamoDbMockClient.on(QueryCommand).resolvesOnce({
+          Items: [
+            {
+              sessionId: { S: "mockSessionId" },
+              clientState: { S: "mockClientState" },
+            },
+          ],
+        });
+
+        const result = await sessionService.getActiveSession("mockSub");
+
+        expect(result.isError).toBe(false);
+        expect(result.value).toEqual({
+          sessionId: "mockSessionId",
+          state: "mockClientState",
+        });
+      });
+    });
+
+    describe("Given the item returned has the attribute redirectUri", () => {
+      it("Returns a success response with the session ID, state and redirectUri as value", async () => {
+        dynamoDbMockClient.on(QueryCommand).resolvesOnce({
+          Items: [
+            {
+              sessionId: { S: "mockSessionId" },
+              clientState: { S: "mockClientSate" },
+              redirectUri: { S: "mockRedirectUri" },
+            },
+          ],
+        });
+
+        const result = await sessionService.getActiveSession("mockSub");
+
+        expect(result.isError).toBe(false);
+        expect(result.value).toStrictEqual({
+          sessionId: "mockSessionId",
+          state: "mockClientSate",
+          redirectUri: "mockRedirectUri",
+        });
+      });
+    });
+  });
+
+  describe("Creates a session", () => {
+    describe("Given there is an unexpected error when creating a session", () => {
+      it("Returns error response", async () => {
+        dynamoDbMockClient.on(PutItemCommand).rejectsOnce("Mock DB Error");
+
+        const result = await sessionService.createSession({
+          state: "mockValidState",
+          sub: "mockSub",
+          client_id: "mockClientId",
+          govuk_signin_journey_id: "mockJourneyId",
+          redirect_uri: "https://mockRedirectUri.com",
+          issuer: "mockIssuer",
+          sessionDurationInSeconds: 3600,
+        });
+
+        expect(result.isError).toBe(true);
+        expect(result.value).toStrictEqual({
+          errorMessage: "Error creating session - Error: Mock DB Error",
+          errorCategory: "SERVER_ERROR",
+        });
+      });
+    });
+
+    describe("Given a session with the same session ID already exists", () => {
+      it("Returns error response", async () => {
+        const mockError = new ConditionalCheckFailedException({
```

Comment:
I don't know why I keep on loving this, but I do

Reply:
It does the job, I guess :)